nifi-nar-bundles/nifi-oraclecdc-bundle/nifi-oraclecdc-processors/src/main/java/org/apache/nifi/processors/oraclecdc/controller/impl/StandardOracleClassLoaderService.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.oraclecdc.controller.impl;
+
+import java.net.MalformedURLException;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnEnabled;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.AbstractControllerService;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.expression.AttributeExpression;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.util.file.classloader.ClassLoaderUtils;
+import org.nifi.oraclecdcservice.api.OracleClassLoaderService;
+
+@Tags({ "dbcp", "jdbc", "database", "connection", "pooling", "store" })
+@CapabilityDescription("Provides a shareable classloader for the Oracle CDC connection pool")
+
+public class StandardOracleClassLoaderService extends AbstractControllerService implements OracleClassLoaderService {
```

Comment:
hi @tpalfy , the purpose is to make sure that we can attach the cdc listener to multiple xstream servers from a single nifi instance. The xstream stuff needs some oracle native libraries. Since native libraries can only be loaded once inside JVM, i had to take this alternative. I do agree it could be utilized elsewhere, especially with the HDFS processors.